client/components/domains/map-domain-step/index.jsx

```diff
@@ -20,16 +20,17 @@ var MapDomainStep = React.createClass( {
 
 	propTypes: {
 		products: React.PropTypes.object.isRequired,
+		initialQuery: React.PropTypes.string,
 		analyticsSection: React.PropTypes.string.isRequired
 	},
 
 	getInitialState: function() {
-		return { searchQuery: '' };
+		return { searchQuery: this.props.initialQuery };
```

Comment:
Your guess is as good as mine :) I don't want to assume anything, so maybe @fabianapsimoes could shed some light here.
Can you tell us why it's needed to store the state (in stores) of the steps during signup? Is it just for navigating forward <=> backward (and, of course, submitting in the end) or is it something else?